source/mqtt/Mqtt5ClientCore.cpp

```diff
@@ -642,6 +609,43 @@ namespace Aws
                 }
             }
 
+            Mqtt5to3AdapterOptions::Mqtt5to3AdapterOptions() {}
+
+            ScopedResource<Mqtt5to3AdapterOptions> Mqtt5to3AdapterOptions::NewMqtt5to3AdapterOptions(
```

Comment:
Debatable: I made the `Mqtt5to3AdapterOptions` as a `ScopedResource`(unique_ptr) so that we could use the aws_allocator to manage the memory of the options. However, I'm also worried it might be over complicated for the case here.